fastcrawler/engine/contracts.py

```diff
@@ -35,24 +39,36 @@ class Response(pydantic.BaseModel):
     url: str | None = None
 
 
-class Request(pydantic.BaseModel):
+@dataclass
+class Request:
     url: str
     proxy: ProxySetting | None = None
     data: dict | str | None = None
     method: str | None = None
     headers: dict | None = None
     cookies: SetCookieParam | None = None
+    sleep_interval: float | None = None
+
+
+@dataclass
+class RequestCycle:
```

Comment:
Request and Response are easy conecepts
but anything Cycle related needs lots of doc strings and comments and examples

Reply:
sure, will do. in another PR